components/feature/addons/src/main/java/mozilla/components/feature/addons/migration/SupportedAddonsChecker.kt

```diff
@@ -160,7 +162,9 @@ internal class SupportedAddonsWorker(
             }
         } catch (exception: Exception) {
             logger.error("An exception happened trying to check for new supported add-ons, re-schedule ${exception.message}", exception)
-            GlobalAddonDependencyProvider.onCrash?.invoke(exception)
+            if (!exception.isNoise()) {
```

Comment:
Let's just add another catch block to handle `IOException`s first. Let's have the type system do this for you. `SocketTimeOutException`s is an IO exception too. So, we don't need this extensions method.

Reply:
👍

Reply:
It will be very convenient to just add an additional catch block, but unfortunately we are wrapping all exceptions that come from [getAddons as AddonManagerException](https://github.com/mozilla-mobile/android-components/blob/e8e4806526aeb895a69e003c7917d3c0ceb986a2/components/feature/addons/src/main/java/mozilla/components/feature/addons/AddonManager.kt#L100), as result we need to unwrap them using `exception.cause`

Reply:
Ah, I missed that! OK, maybe just call it `exception.shouldReport()`?

Reply:
Sounds good! :)